File: CodeKata/QueueUsingStacks/Queue/Queue.cs
﻿using System;
using System.Collections;

namespace Queue
{
    public class Queue
    {
        Stack s0;
        Stack s1;

        public Queue()
        {
            s0 = new Stack();
            s1 = new Stack();
        }

        public void Enqueue(char c)
        {
            s0.Push(c);
        }

        public char Dequeue()
        {
            if(s1.Count == 0)
            {
                
            }
            return '0';
        }

        public char Peek()
        {
            return '0';
        }

        public bool Contains(char c)
        {
            if(s0.Contains(c))
            {
                return true;
            }
            if(s1.Contains(c))
            {
                return true;
            }
            return false;
        }

        public int Size()
        {
            return s0.Count + s1.Count;
        }
    }
}

File: solutions/CPP/QueueUsingStacks/src/queue.cpp
#include "queue.h"

void Queue::update()
{
	if(b.size() == 0)
	{
		while(a.size() > 0)
		{
			int d = a.top();
			a.pop();
			b.push(d);
		}
	}
}

int Queue::top()
{
	update();
	return b.top();
}

void Queue::enqueue(int i)
{
	a.push(i);
}

int Queue::dequeue()
{
	update();
	int r = b.top();
	b.pop();
	return r;
}

int Queue::size()
{
	return a.size() + b.size();
}
